beacon-chain/blockchain/process_block.go

```diff
@@ -134,6 +135,9 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 	}
 	isValidPayload, err := s.notifyNewPayload(ctx, postStateVersion, postStateHeader, signed)
 	if err != nil {
+		if IsInvalidBlock(err) && InvalidBlockLVH(err) != [32]byte{} {
+			return s.reportInvalidBlock(ctx, blockRoot, parentRoot, InvalidBlockLVH(err))
```

Comment:
I think you are losing the error message here

Reply:
It will return the error `ErrInvalidPayload` which contains the relevant message

Reply:
You are right. I missed that field